### ISSUE TITLE
Explicit transaction spanning multiple flush() calls

### ISSUE THREAD

**Issue by jkavalik** [nextras/orm#191]
Currently even explicitly started transaction is commited during any `flush()` or `xAndFlush()` call which quite surprised me. Is there any way to not commit during the `flush()` and let the commit/rollback on the user? Or transaction stacking where you have to commit as many times as startTransaction for the commit to actually happen (and any rollback along the way would probably have to change the transaction to `rollbackOnly`)?

**Comment by hrach** [comment]
Yes, this seems to be very unpredictable and unusable when you want to manipulate with transation ourself.

**Comment by hrach** [comment]
@jkavalik could you please provide some sample of code with description what is current status and how would you expect it should work?

**Comment by hrach** [comment]
I have reviews the code and this issue and came to conclusion:

- the main problem was missing support for nested transactions in Nextras Dbal, which was already implemented.
- Nextras Orm shouldn't be aware of something like manually started transactions in Dbal layers, still, Orm should always commit such transactions which were opened by Orm. 
- Nextras Orm will still open a transaction when persisting changes and flush will still commit such transaction, however, since nextras/dbal#15 the code which opens explictly new transaction should work ok:

```php
$connection->transactional(function() {
    $orm->persistAndFlush(...);
    ...
    $orm->persistAndFlush(...);
});
```

the aformentioned code will now result in correct transaction nesting:
1. begin transaction
2. savepoint xx
3. release savepoint xx
4. savepoint yy
5. release savepoint yy
6. commit

So, this is fixed in Nextras Dbal 3.0.
